geomesa-filter/src/main/scala/org/locationtech/geomesa/filter/function/BinaryOutputEncoder.scala

```diff
@@ -0,0 +1,226 @@
+
+  implicit val ordering = new Ordering[ValuesToEncode]() {
+    override def compare(x: ValuesToEncode, y: ValuesToEncode) = x.dtg.compareTo(y.dtg)
+  }
+
+  /**
+   * Encodes a feature collection to bin format. Features are written to the output stream.
+   *
+   * @param fc
+   * @param output
+   * @param dtgField
+   * @param trackIdField
+   * @param labelField
+   * @param latLon
+   * @param axisOrder
+   * @param sort
+   */
+  def encodeFeatureCollection(
+      fc: SimpleFeatureCollection,
+      output: OutputStream,
+      dtgField: String,
+      trackIdField: Option[String],
+      labelField: Option[String] = None,
+      latLon: Option[(String, String)] = None,
+      axisOrder: AxisOrder = LatLon,
+      sort: Boolean = false): Unit = {
+
+    val sft = fc.getSchema
+    val isLineString = sft.getGeometryDescriptor.getType.getBinding == classOf[LineString]
+
+    // validate our input data
+    validateDateAttribute(dtgField, sft, isLineString)
+    validateLabels(trackIdField, labelField, sft)
+    validateLatLong(latLon, sft)
+
+    val sysTime = System.currentTimeMillis
+    val dtgIndex = sft.indexOf(dtgField)
+
+    // get date function based
+    val getDtg: (SimpleFeature) => Long = (f) => {
+      val date = f.getAttribute(dtgIndex).asInstanceOf[Date]
+      if (date == null) sysTime else date.getTime
+    }
+    // for line strings, we need an array of dates corresponding to the points in the line
+    val getLineDtg: (SimpleFeature) => Array[Long] = (f) => {
+      val dates = f.getAttribute(dtgIndex).asInstanceOf[java.util.List[Date]]
+      if (dates == null) Array.empty else dates.map(_.getTime).toArray
+    }
+
+    // get lat/lon as floats
+    val getLatLon: (SimpleFeature) => (Float, Float) =
+      latLon.map { case (lat, lon) =>
+        // if requested, export arbitrary fields as lat/lon
+        val latIndex = sft.indexOf(lat)
+        val lonIndex = sft.indexOf(lon)
+        val binding = sft.getType(latIndex).getBinding
+        if (binding == classOf[java.lang.Float]) {
+          (f: SimpleFeature) =>
+            (f.getAttribute(latIndex).asInstanceOf[Float], f.getAttribute(lonIndex).asInstanceOf[Float])
+        } else if (binding == classOf[java.lang.Double]) {
+          (f: SimpleFeature) =>
+            (f.getAttribute(latIndex).asInstanceOf[Double].toFloat, f.getAttribute(lonIndex).asInstanceOf[Double].toFloat)
+        } else {
+          (f: SimpleFeature) =>
+            (f.getAttribute(latIndex).toString.toFloat, f.getAttribute(lonIndex).toString.toFloat)
+        }
+      }.getOrElse {
+        // default is to use the geometry
+        // depending on srs requested and wfs versions, axis order can be flipped
+        axisOrder match {
+          case LatLon => (f) => {
+            val geom = f.getDefaultGeometry.asInstanceOf[Geometry].getInteriorPoint
+            (geom.getX.toFloat, geom.getY.toFloat)
```

Comment:
maybe extract function: 

  val pointToXY: Point => (Float, Float) = 
      p => (p.getX.toFloat, p.getY.toFloat)

and use thoroughout. For the YX case you can use the same function and then call swap on the tuple